test/int/framework/frameworkCommonTests.ts

```diff
@@ -125,6 +126,30 @@ export class FrameworkTestSuite {
             await debugClient.continueRequest();
         });
     }
+
+    /**
+     * A generic breakpoint test. This can be used for many different types of breakpoint tests with the following structure:
+     *
+     * 1. Wait for the page to load by waiting for the selector: `waitSelectorId`
+     * 2. Set a breakpoint at `bpLabel`
+     * 3. Execute a trigger event that should cause the breakpoint to be hit using the function `trigger`
+     * 4. Assert that the breakpoint is hit on the expected location, and continue
+     *
+     * @param waitSelectorId
+     * @param bpLabel
+     * @param trigger
+     */
+    genericBreakpointTest(description: string, waitSelectorId: string, bpLabel: string, trigger: (page: puppeteer.Page) => Promise<void>) {
+        return puppeteerTest(`${this.frameworkName} - ${description}`, this.suiteContext, async (context, page) => {
+            const location = context.breakpointLabels.get(bpLabel);
+            await page.waitForSelector(`#${waitSelectorId}`);
```

Comment:
I'd personally add the # on the caller. It gives you the option to use other selectors that are not based on #

Reply:
Good point. Done.